server/api/email.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+const Boom = require('boom');
+const Proposal = require('../models/proposal');
+const Feedback = require('../models/feedback');
+const User = require('../models/user');
+const Mailer = require('../mailer');
+const Config = require('../../config');
+var addDays = require('date-fns/addDays');
+var format = require('date-fns/format');
+
+const register = function (server, options) {  
+
+  server.route({
+    method: 'POST',
+    path: '/api/email/',
+    options: {
+      auth: {
+        strategies: ['simple', 'session'],
+        scope: ['reviewer', 'root', 'coordinator', 'chair']
+      }      
+    },      
+    handler: async function (request, h) {
+    
+      let template = request.payload.templateName;
+      let subject;  
+      let emailOptions;         
+      let emailTemplateData;
+      
+      /*
+        Case for when Chair assigns reviewers to a proposal. Emails are sent to all selected reviewers
+      */
+        if(template === 'reviewers-to-review-proposal'){
+            subject = 'You have been assigned to review a proposal';
+            const proposalDoc = await Proposal.findById(request.payload.proposalId); 
+            const reviewerIds = proposalDoc.reviewerIds;
+            var reviewerEmails = [];
+            for(let id in reviewerIds){
+                const reviewer = await User.findById(reviewerIds[id]);
+                reviewerEmails.push(reviewer.email);
+            }
+            const today = new Date();
+            const laterDate = addDays(today, 28);
+            const daysAfter = format(laterDate, 'MM/dd/yyyy');
+            emailOptions = {
+                subject: subject,
+                to: {
+                    address: reviewerEmails
+                },
+                cc: Config.get('/EmailList/ccAddress')
+            };
+            emailTemplateData = {
+                fileName: proposalDoc.fileName, // only a single file name
+                daysAfter: daysAfter.toString()
+            };
+        }
+      
+      /*
+        Case for when Chair finalizes decision for a proposal. 
+      */ 
+        if(template === 'chair-finalized-decision'){
+            subject = 'BROC Chair has finalized decision';
+            const proposalDoc = await Proposal.findById(request.payload.proposalId);
```

Comment:
don't need this! see my comment above regarding defining proposalDoc outside if statements.